src/pytest_rerunfailures.py

```diff
@@ -548,9 +548,24 @@ def pytest_runtest_protocol(item, nextitem):
         item.ihook.pytest_runtest_logstart(nodeid=item.nodeid, location=item.location)
         reports = runtestprotocol(item, nextitem=nextitem, log=False)
 
+        # Check all reports to see if any rerun is needed
+        # (So teardown report is checked before processing call (test))
+        except_found = False
+        for r in reports:
+            rerun_except_errors = _get_rerun_filter_regex(item, "rerun_except")
+            except_found = rerun_except_errors and _matches_any_rerun_except_error(
+                rerun_except_errors, r
+            )
+            if except_found:
+                break
+
+        should_not_rerun = True
+        for r in reports:
```

Comment:
The for loop in line 567 uses `for report in reports`, so I'd like it here to for consistency.

Reply:
thanks, fixed and also fix except reports loop to break after finding first except match so later report without except match will not override it (This fix all the test failures)